### PR TITLE
chore: [CDS-86275]: onboard GitX to template studio

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Add support for GitX parameters in new yaml simplification studio
- Handle status and error scenarios for openAPI response
- Move execution labels behind CDS_YAML_SIMPLIFICATION FF

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts [hunk 1]
```diff
@@ -182,7 +183,7 @@ export function useSaveTemplateY1({
           break
       }
     } catch (err) {
-      throw response
+      throw response || err
```
Comment (PratyushGar) [inline src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:186]: The 400 error response from API is getting propagated to catch block as err.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts (hunk 1 above)
Comment (mayankVermaHarness) [reply src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:186]: Catching and re-throwing from catch doesn't seem like a good pattern.

Is there a higher level catch for this re-throw?

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts [hunk 2]
```diff
@@ -313,7 +316,7 @@ export function useSaveTemplateY1({
             break
         }
       } catch (err) {
-        throw response
+        throw response || err
```
Comment (mayankVermaHarness) [inline src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:319]: Same comment as https://github.com/harness/harness-core-ui/pull/19728/files#r1445851261

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts (hunk 1 above)
Comment (PratyushGar) [reply src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:186]: Yes, parent catch is present https://github.com/harness/harness-core-ui/blob/84831c43a8f216fec560b92552f8d69a76e2d5be/src/modules/10-common/modals/SaveToGitDialog/useSaveToGitDialog.tsx#L382, which handles all error scenario to render detailed ErrorSummary component

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts (hunk 2 above)
Comment (PratyushGar) [reply src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:319]: Yes, parent catch is present https://github.com/harness/harness-core-ui/blob/84831c43a8f216fec560b92552f8d69a76e2d5be/src/modules/10-common/modals/SaveToGitDialog/useSaveToGitDialog.tsx#L382, which handles all error scenario to render detailed ErrorSummary component

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts [hunk 3]
```diff
@@ -275,6 +277,7 @@ export function useSaveTemplateY1({
         body: {
           template_yaml: templateYaml,
           comments,
+          ...(isNewTemplateIdentifierCreation ? { isNewTemplate: true } : {}),
```
Comment (samarth-sharma31) [inline src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:280]: Please test these scenarios as well ->
1. Save Template
2. Save as new version
3. save as new template
4. Update template

Comment (PratyushGar) [reply src/modules/72-templates-library/y1/utils/useSaveTemplateY1.ts:280]: Yes, verified for all these scenarios as well

**Review by samarth-sharma31 — APPROVED** [review]

**Pull request merged**